src/world/message_server.cpp

```diff
@@ -163,6 +205,15 @@ void message_server_parse(MSGSERVTYPE type, zmq::message_t* extra, zmq::message_
             // no op
             break;
         }
+        case MSG_MAP2WORLD_REGIONAL_EVENT:
+        {
+            uint8*      data    = (uint8*)extra->data();
+            const uint8 subType = ref<uint8>(data, 0);
+
+            std::vector<uint8> bytes(data, data + extra->size());
+            regionalMsgHandlers[(REGIONALMSGTYPE)subType]->handleMessage(bytes, from_ip, from_port);
```

Comment:
👍

Reply:
In the interest of safety, I'd look into not using operator[] and using .get() instead, and wrapping this in a try/catch block -> because world-server needs to be bulletproof. 

If an unvalidated subType comes in, operator[] access in the map will create a new shared_ptr, which we immediately call handleMessage on. I'm not sure if this would be a nullptr dereference or not, but best not to risk it

Reply:
https://en.cppreference.com/w/cpp/memory/unique_ptr/operator*

Turns out this sort of access is undefined behaviour, lets avoid that :D

Reply:
So I understand. Avoid using ->get() or avoid accessing without checking first?

Reply:
```cpp
try
{
    auto& handler = regionalMsgHandlers.get((REGIONALMSGTYPE)subType); // or is this a shared ptr? I don't remember
    handler->handleMessage(...);
}
catch (the possible exception from get() throwing)
{
...
}
```

Reply:
The idea is to use get, because get will throw if the key isn't there. If you use operator[] and it isn't there, it'll create an entry (an empty shared ptr). Accessing members of the underlying class of an empty shared ptr is undefined behaviour (it might work, it might crash, we can't know)

Reply:
gotcha thanks for explaining.

Reply:
Done